### PR TITLE
Sync-related logging #314

### PR DESCRIPTION
**Pull request by taychunmei**
#314

### CODE REVIEW

File: src/model/Model.java [hunk 1]
```diff
@@ -296,8 +297,10 @@ public void updateCachedIssue(TurboIssue issue){
 		TurboIssue tIssue = getIssueWithId(issue.getId());
 		if(tIssue != null){
 			tIssue.copyValues(issue);
-		}else{
+			logger.info("Updated issue: " + issue.getId());
```
Comment (damithc) [inline src/model/Model.java]: This kind of detailed logs can be done at `debug` level. Furthermore, try to record what exactly was changed (i.e. the new values). Can be helpful in debugging cases of 'missing labels' etc.

File: src/storage/DataCacheFileHandler.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ public void setIssues(List<TurboIssue> issues) {
 	}
 	
 	public void writeToFile(String repoIdString, String issuesETag, String collabsETag, String labelsETag, String milestonesETag, String issueCheckTime, ObservableList<TurboUser> collaborators, ObservableList<TurboLabel> labels, ObservableList<TurboMilestone> milestones, ObservableList<TurboIssue> issues) {
-		logger.info("About to write to file...");
+		logger.info("About to write to file for repo: " + repoIdString);
```
Comment (damithc) [inline src/storage/DataCacheFileHandler.java]: Are you logging the 'last checked time' property somewhere?

**Comment by damithc** [comment]
Added a couple of comments after a quick look. @dariusf to review in detail.

File: src/model/Model.java [hunk 3]
```diff
@@ -384,6 +387,15 @@ private void updateCachedList(List list, List newList, String repoId){
...
 	        	list.removeAll(removed);
+  	
+	        	Listable listItem = (Listable)newList.get(0);
+	        	if (listItem.getClass() == TurboMilestone.class) {
+	        		logNumOfUpdates(newList, "milestone(s)");
+	        	} else if (listItem.getClass() == TurboLabel.class) {
+	        		logNumOfUpdates(newList, "label(s)");
+	        	} else if (listItem.getClass() == TurboUser.class) {
+	        		logNumOfUpdates(newList, "collaborator(s)");
+	        	}
```
Comment (dariusf) [inline src/model/Model.java]: Using `instanceof` is probably better, as it's less brittle (it'll still work if we subclass any of the `Turbo*` classes).

**Pull request merged**